ddtrace/tracer/telemetry.go

```diff
@@ -70,9 +73,11 @@ func startTelemetry(c *config) {
 			telemetry.Configuration{Name: fmt.Sprintf("sr_%s_(%s)_(%s)", rule.ruleType.String(), service, name),
 				Value: fmt.Sprintf("rate:%f_maxPerSecond:%f", rule.Rate, rule.MaxPerSecond)})
 	}
-	telemetry.GlobalClient.Start(telemetryConfigs)
+	telemetry.GlobalClient.ProductStart(telemetry.NamespaceTracers, telemetryConfigs)
 }
 
+// stopTelemetry is called when the tracer stops. Because appsec stops when the tracer stops,
+// we sent a ProductStop message for appsec.
 func stopTelemetry() {
-	telemetry.GlobalClient.Stop()
```

Comment:
Currently the `app-closing` event is sent after the tracer stops. However, we only want the `app-closing` event to be sent when a SIGINT or SIGQUIT signal is intercepted (TODO, for another PR). Though this would miss several cases of an app closing gracefully, it is the expected behavior from the telemetry API docs. 

Instead, on tracer.Stop we call `ProductStop` for appsec, since appsec stops when the tracer stops

Reply:
It took me a while of looking at the code and various code comments to understand this piece of the code, despite the comment. I think the comment you have is good, but expand it just a bit more by noting that tracing is assumed to always be enabled by instrumentation telemetry, and it isn't considered a standalone product to be turned on/off, therefore no ProductStop message is required for `telemetry.NamespaceTracers`